src/python/grpcio/commands.py

```diff
@@ -58,6 +63,117 @@ class CommandError(Exception):
 
 
+# TODO(atash): Remove this once PyPI has better Linux bdist support
+def _linux_target_egg_name(grpc_version):
+  """Get gRPC's custom made-up target names for Linux."""
+  assert sys.platform.startswith('linux')
+  version = '.'.join(map(str, sys.version_info[0:2]))
+  architecture = platform.machine()
+  flavor = 'ucs2' if sys.maxunicode == 65535 else 'ucs4'
+  return (
+      'grpcio-{grpc_version}-py{version}-linux-{architecture}-{flavor}.egg'
+        .format(grpc_version=grpc_version,
+                version=version,
+                architecture=architecture,
+                flavor=flavor))
+
+
+class Install(install.install):
+  """Custom Install command for gRPC Python.
+
+  This is for bdist shims and whatever else we might need a custom install
+  command for.
+  """
+
+  user_options = install.install.user_options + [
+      # TODO(atash): remove this once manylinux gets on PyPI
+      ('use-linux-bdist', None,
+       'Whether to retrieve a binary for Linux instead of building from '
+       'source.'),
+  ]
+
+  def initialize_options(self):
+    install.install.initialize_options(self)
+    self.use_linux_bdist = False
+
+  def finalize_options(self):
+    install.install.finalize_options(self)
+
+  def run(self):
+    if self.use_linux_bdist:
+      try:
+        egg_path = self._get_linux_bdist_egg()
+      except CommandError as error:
+        sys.stderr.write(
+            '\nWARNING: Failed to acquire grpcio prebuilt binary:\n'
+            '{}.\n\n'.format(error.message))
+        raise
+      try:
+        self._run_bdist_retrieval_install(egg_path)
+      except KeyboardInterrupt:
+        raise
+      except Exception as error:
+        # if anything else happens (and given how there's no way to really know
+        # what's happening in setuptools here, I mean *anything*), warn the user
+        # and fall back to building from source.
+        sys.stderr.write(
+            '{}\nWARNING: Failed to install grpcio prebuilt binary.\n\n'
+                .format(traceback.format_exc()))
+        install.install.run(self)
+    else:
+      install.install.run(self)
+
+  # TODO(atash): Remove this once PyPI has better Linux bdist support
+  def _run_bdist_retrieval_install(self, bdist_egg):
+    easy_install = self.distribution.get_command_class('easy_install')
+    easy_install_command = easy_install(
+        self.distribution, args='x', root=self.root, record=self.record,
+    )
+    easy_install_command.ensure_finalized()
+    easy_install_command.always_copy_from = '.'
+    easy_install_command.package_index.scan(glob.glob('*.egg'))
+    arguments = [bdist_egg]
+    if setuptools.bootstrap_install_from:
+      args.insert(0, setuptools.bootstrap_install_from)
+    easy_install_command.args = arguments
+
+    easy_install_command.run()
+    setuptools.bootstrap_install_from = None
+
+  # TODO(atash): Remove this once PyPI has better Linux bdist support
+  def _get_linux_bdist_egg(self):
```

Comment:
By changing "self" to "distribution" (or "version"), you can pull this helper behavior out of the class without increasing its parameter count.